### PR TITLE
Added Blocking Room Bookings.

### PR DESCRIPTION
**Pull request by pavitgopal**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -50,6 +50,16 @@
...
+  create_table 'room_blocks', force: :cascade do |t|
+    t.integer 'room_id', null: false
+    t.integer 'time_slot_id', null: false
+    t.boolean 'is_blocked', default: true
+    t.datetime 'created_at', null: false
+    t.datetime 'updated_at', null: false
+    t.index ['room_id'], name: 'index_room_blocks_on_room_id'
+    t.index ['time_slot_id'], name: 'index_room_blocks_on_time_slot_id'
+  end
+
```
Comment (Naviunni) [inline db/schema.rb]: Where is the migration for this change?

**Review by Naviunni — CHANGES_REQUESTED** [review]

File: db/schema.rb (hunk 1 above)
Comment (pavitgopal) [reply db/schema.rb]: we decided to use room_bookings instead, because I didn't realize that we had the ability to block off times and rooms through that.

**Review by wahibkapdi — DISMISSED** [review]

File: app/controllers/room_bookings_controller.rb [hunk 2]
```diff
@@ -19,14 +19,28 @@ def index
       section: [:course],
       time_slot: {},
       instructor: {}
-    ).where(rooms: { schedule_id: }, time_slots: { day: @active_tab })
+    ).where(rooms: { schedule_id: @schedule.id }, time_slots: { day: @active_tab })
```
Comment (Naviunni) [inline app/controllers/room_bookings_controller.rb:21]: we already have schedule_id saved, so why are we getting the schedule and then its ID?

File: app/controllers/room_bookings_controller.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 
-# Room Bookings Controller
 class RoomBookingsController < ApplicationController
+  before_action :set_schedule
+
   def index
-    schedule_id = params[:schedule_id]
-    @schedule = Schedule.find(params[:schedule_id])
+    # @schedule = Schedule.find(params[:schedule_id])
```
Comment (Naviunni) [inline app/controllers/room_bookings_controller.rb]: Remove comment

File: app/views/room_bookings/index.html.erb [hunk 4]
```diff
@@ -122,10 +128,14 @@
                 }
             });
 
+            // Fix schedule id now.
```
Comment (Naviunni) [inline app/views/room_bookings/index.html.erb]: Are these to-dos? if fixed, pls remove the comment

File: features/step_definitions/room_booking_management_steps.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  ).is_available).to eq(false)
+end
+Then('the booking in {string} {string} at {string} {string} should be available') do |building, room, start, days|
+  expect(RoomBooking.find_by(
+    room: Room.find_by(schedule_id: @schedule.id, building_code: building, room_number: room),
+    time_slot: TimeSlot.find_by(start_time: start, day: days)
+  ).is_available).to eq(true)
+end
+
+Then('overlapping bookings should also be unavailable') do
```
Comment (Naviunni) [inline features/step_definitions/room_booking_management_steps.rb:43]: Remove unused step

File: app/controllers/room_bookings_controller.rb (hunk 2 above)
Comment (Kanchuking1) [reply app/controllers/room_bookings_controller.rb:21]: We don't have schedule_id anymore, it's been deleted to use the common functionality.

File: features/step_definitions/room_booking_management_steps.rb (hunk 5 above)
Comment (Kanchuking1) [reply features/step_definitions/room_booking_management_steps.rb:43]: This needs to be implemented in the later MRs.

**Review by Naviunni — APPROVED** [review]

**Pull request merged**